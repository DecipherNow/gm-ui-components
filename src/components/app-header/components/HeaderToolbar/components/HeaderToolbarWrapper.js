import styled from "styled-components";

import { readableColor, transparentize } from 'polished';

import { spacingScale } from "style/styleFunctions";
import { FONT_SIZE_BASE, COLOR_BACKGROUND_C } from "style/styleVariables";

const APP_TOOLBAR_HEIGHT = spacingScale(4);
const APP_TOOLBAR_FONT_SIZE = FONT_SIZE_BASE;
const APP_TOOLBAR_BACKGROUND_COLOR = "transparent";
const APP_TOOLBAR_TEXT_COLOR = readableColor(COLOR_BACKGROUND_C);

const HeaderToolbarWrapper = styled.nav`
  background-color: ${APP_TOOLBAR_BACKGROUND_COLOR};
  color: ${APP_TOOLBAR_TEXT_COLOR};
  flex: 0 0 ${APP_TOOLBAR_HEIGHT};
  font-size: ${APP_TOOLBAR_FONT_SIZE};
  background-image: linear-gradient(to top, ${transparentize(1, COLOR_BACKGROUND_C)}, ${COLOR_BACKGROUND_C});
  padding: 0 ${spacingScale(2)};
  align-items: center;
  display: flex;
  flex-wrap: nowrap;
  background-size: 100% 100%;
  background-repeat: no-repeat;
`;

export default HeaderToolbarWrapper;
