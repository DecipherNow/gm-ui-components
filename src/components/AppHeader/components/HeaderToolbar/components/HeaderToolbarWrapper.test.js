import React from "react";
import { shallow } from "enzyme";
import HeaderToolbarWrapper from "./HeaderToolbarWrapper";

describe("HeaderToolbarWrapper", () => {
  it("matches snapshot", () => {
    const aHeaderToolbarWrapper = shallow(<HeaderToolbarWrapper />);
    expect(aHeaderToolbarWrapper).toMatchSnapshot();
  });
});
