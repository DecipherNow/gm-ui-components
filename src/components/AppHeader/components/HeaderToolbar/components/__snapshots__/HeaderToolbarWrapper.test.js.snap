// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderToolbarWrapper matches snapshot 1`] = `
.c0 {
  background-image: linear-gradient(to top,transparent,#000);
  background-color: transparent;
  color: #fff;
  -webkit-flex: 0 0 32px;
  -ms-flex: 0 0 32px;
  flex: 0 0 32px;
  font-size: 14px;
  padding: 0 16px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  background-size: 100% 100%;
  background-repeat: no-repeat;
}

<nav
  className="c0"
/>
`;
